core/src/main/java/com/valtech/aem/saas/core/http/client/DefaultSearchAdminRequestExecutorService.java

```diff
@@ -0,0 +1,124 @@
+    }
+
+    @Override
+    public Optional<SearchResponse> execute(@NonNull SearchRequest searchRequest) {
+        return new SearchRequestExecutor(httpClient).execute(searchRequest);
+
+    }
+
+    @Activate
+    @Modified
+    private void activate(Configuration configuration) {
+        this.configuration = configuration;
+        initHttpClient();
+    }
+
+    private void initHttpClient() {
+        httpClient = SearchHttpClientFactory.builder()
+                                            .httpClientBuilderFactory(httpClientBuilderFactory)
+                                            .httpConnectionTimeout(getSearchConnectionConfig().getHttpConnectionTimeout())
+                                            .httpSocketTimeout(getSearchConnectionConfig().getHttpSocketTimeout())
+                                            .jwtAuthenticationEnabled(configuration.searchAdminRequestExecutorService_jwtAuthentication_enable())
+                                            .jwtToken(configuration.searchAdminRequestExecutorService_jwtAuthentication_token())
+                                            .basicAuthenticationEnabled(configuration.searchAdminRequestExecutorService_basicAuthentication_enable())
+                                            .baseUrl(configuration.searchAdminRequestExecutorService_baseurl())
+                                            .basicAuthenticationUser(configuration.searchAdminRequestExecutorService_basicAuthentication_user())
+                                            .basicAuthenticationUser(configuration.searchAdminRequestExecutorService_basicAuthentication_password())
+                                            .ignoreSslEnabled(getSearchConnectionConfig().isIgnoreSslEnabled())
+                                            .httpMaxTotalConnections(getSearchConnectionConfig().getHttpMaxTotalConnections())
+                                            .httpMaxConnectionsPerRoute(getSearchConnectionConfig().getHttpMaxConnectionsPerRoute())
+                                            .build()
+                                            .create();
+    }
+
+    @Deactivate
+    private void deactivate() {
+        IOUtils.closeQuietly(httpClient, e -> log.error("Could not close client.", e));
+    }
+
+    @ObjectClassDefinition(name = "Search as a Service - Search Admin Request Executor Service Configuration",
+                           description = "URL and authentication details for connect to Search Admin Endpoint.")
+    public @interface Configuration {
+
+        String DEFAULT_WEB_SERVICE_URL = "https://search-admin.infocentric.swiss/admin";
+        boolean DEFAULT_BASIC_AUTHENTICATION_ENABLE = false;
+        boolean DEFAULT_JWT_AUTHENTICATION_ENABLE = false;
+
+        @AttributeDefinition(name = "Base URL",
+                             description = "The protocol + url for the search service") String searchAdminRequestExecutorService_baseurl() default DEFAULT_WEB_SERVICE_URL; // NOSONAR
```

Comment:
could you add a linebreak after the annotation definitions? this for all occurances in here. It should improve readability
```
@AttributeDefinition(name = "Base URL",
                             description = "The protocol + url for the search service") 
String searchAdminRequestExecutorService_baseurl() default DEFAULT_WEB_SERVICE_URL; // NOSONAR
```